### PR TITLE
Feature/major update

### PR DESCRIPTION
**Pull request by ehmicky**
This is a proposal for a major update of `gulp-yaml`. It includes both changes from the `refactor` branch and the two currently opened pull requests: #3 and #7.

# User features

Add `options.ext` to specify the filename extension (defaults to `.json`).

Add `options.json` and `options.onWarning` from `js-yaml`.

Add support for `file.contents` being `null`.

# Documentation

Improve `README.md`.

# Dependencies

Remove `gulp-util` deprecated dependency.

Upgrade `js-yaml` from `3.4.3` to `3.11.0`

Upgrade `bufferstreams` from `1.1.0` to `2.0.0`

Add `package-lock.json`.

# Linting

Add linting with ESLint. It is based on [Standard JavaScript](https://standardjs.com/) with two rules exceptions to keep the current coding style untouched. 

# Refactoring

Major code refactoring.

# Note

I see there has been only one commit since November 2015. If you need someone to maintain this project, I can volunteer!

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+extends:
+- standard
+
+root: true
+
+env:
+  mocha: true
+
+rules:
```
Comment (crissdev) [inline .eslintrc.yml]: Let's stick to standardjs - I don't mind removing the commas or the extra space before functions's parens.

File: index.js [hunk 2]
```diff
@@ -1,99 +1,109 @@
 'use strict';
 
-var replaceExt    = require('replace-ext');
-var through       = require('through2');
-var yaml          = require('js-yaml');
-var xtend         = require('xtend');
+var yaml = require('js-yaml');
```
Comment (crissdev) [inline index.js]: Maybe using const/let instead of var, where possible?

**Comment by crissdev** [comment]
> I see there has been only one commit since November 2015. If you need someone to maintain this project, I can volunteer!

Your help is welcome. I've added you as a collaborator 😄 Thanks.

File: index.js (hunk 2 above)
Comment (ehmicky) [reply index.js]: Yes, absolutely. I only stuck to `var` because the code base only used `var` before refactoring.

File: .eslintrc.yml (hunk 1 above)
Comment (ehmicky) [reply .eslintrc.yml]: Alright, I'll fix that. Just wanted not to change your coding style, which is why I added those two exceptions.

**Comment by ehmicky** [comment]
Alright fixed everywhere. Waiting for your green light.

About being added as a collaborator: great! I'll probably submit new PRs in coming days or weeks.

**Pull request merged**